### DISCUSSION TITLE
`cabalFlag` of type `string`, instead of `boolean`

### DISCUSSION THREAD

**Discussion by peterbecich** [srid/haskell-flake#264] [Q&A]
Can `haskell-flake` set a Cabal flag which is a `string`?
i.e. there is an issue with `hackage-server` which requires the flag `--write-ghc-environment=always`: https://github.com/isovector/type-errors/issues/9

In this `haskell-flake` https://github.com/peterbecich/hackage-server/blob/enable-check/flake.nix I am trying to specify
```
hackage-server.cabalFlags.write-ghc-environment = "always";
```

Reading this it appears `strings` are not supported: https://github.com/srid/haskell-flake/blob/9173cc45aeb72b7e7adfe0e5a53a425fe439e3ca/nix/modules/project/settings/all.nix#L203

Thanks

**Comment by srid** [comment]
I think you are conflating two things? cabal flags vs cabal build arguments.

You probably want `extraBuildFlags`

https://github.com/srid/haskell-flake/blob/9173cc45aeb72b7e7adfe0e5a53a425fe439e3ca/nix/modules/project/settings/all.nix#L183-L189

**Reply by peterbecich** [reply]
You're right, I was unaware of `extraBuildFlags`. However, it rejects this:
```
hackage-server.extraBuildFlags = ["--write-ghc-environment=always"];
```

Result is:
```
       > Running phase: buildPhase
       > unrecognized 'build' option `--write-ghc-environment=always'
```

This works fine:
```
cabal build --write-ghc-environment=always
```


`--verbose` is accepted:
```
hackage-server.extraBuildFlags = ["--verbose"];
```

**Reply by srid** [reply]
How do people normally pass `--write-ghc-environment-files` when using [plain nixpkgs](https://nixos.asia/en/nixify-haskell-nixpkgs)? 

If we can answer that, I can implement it as an option in `haskell-flake`.

**Reply by peterbecich** [reply]
Looking, will follow up, thanks

**Reply by shivaraj-bh** [reply]
You could try `extraConfigureFlags = [ “--write-ghc-environment=always” ];`

**Reply by peterbecich** [reply]
good idea, thanks: https://github.com/haskell/hackage-server/blob/76b365c15f2ccce0cd42317f2e6b118e0873141d/flake.nix#L30

However, it rejects that:
```
.
> unrecognized 'configure' option `--write-ghc-environment=always'
.
```

**Reply by peterbecich** [reply]
This issue is not important, it is not worth your time. I am trying to get the `DocTests` to succeed for Hackage Server. However, `write-ghc-environment` will probably not solve it: https://github.com/haskell/hackage-server/issues/1197#issuecomment-1987030619

Upgrading NixPkgs to GHC 9.8 is likely to solve it.
```
basePackages = pkgs.haskell.packages.ghc98;
```
That is blocked on other dependencies in NixPkgs.
